### ISSUE TITLE
[Poll] Time for v1.0.0?

### ISSUE THREAD

**Issue by xaviergonz** [xaviergonz/mobx-keystone#38]
After the latest PR to remove "$" from paths I don't foresee big changes to the API for the time being. Also most issues seem to have been resolved now. Do you think it is time to brand it as v1.0.0?

**Comment by xaviergonz** [comment]
Well, actually there might be one more to add a better way to do async actions like in here:
https://github.com/mobxjs/mobx/pull/2118
but it depends on this mobx PR being accepted:
https://github.com/mobxjs/mobx/pull/2117

**Comment by sisp** [comment]
I have been thinking whether it makes sense to add a utility function `findParentPath` in addition to `findParent` which also returns the path of the found parent like in `RootPath`.

**Comment by xaviergonz** [comment]
Wouldn't that be the same as running getRootPath over the result?

**Comment by sisp** [comment]
No, I meant that `findParentPath` may return the parent object and the path from the parent to the object, similar to `getParentPath` but with depth ≥ 1.

**Comment by xaviergonz** [comment]
Ah, sounds good :) https://github.com/xaviergonz/mobx-keystone/pull/43

**Comment by sisp** [comment]
About backrefs: Shouldn't the return type of `getRefsResolvingTo` be a readonly `ObservableSet<Ref<T>>`? (Not sure if that's possible by the way because MobX doesn't seem to have a `ReadonlyObservableSet` type ...)

**Comment by xaviergonz** [comment]
It should, but there are no readonly versions of observable stuff in mobx types :(

**Comment by terrysahaidak** [comment]
Didn't find any mention of custom types in docs, @xaviergonz.
Simple case: My server responds with a timestamp as a date, I'm converting it into regular Date object in js, and converting back into the timestamp for snapshots. In MST I've been using `types.custom` to address it. Is there something similar in mobx-keystone?

**Comment by xaviergonz** [comment]
There's none at the moment, but in theory you could have something like
```ts
class ... extends Model({timestamp: prop<number>()}) {
  @computed
  get date() { return new Date(this.timestamp) }

  @modelAction
  setDate(date: Date) { this.timestamp = +date }
}
```

I think there could be a way to extract this into a composable pattern
(not working code):

```ts
const asDate = (getter: () => number, setter: (ts: number) => void) => {
  return {
    get: computed(() => new Date(getter())),
    set: modelAction((d: Date) => { setter(+d) })
  }
}

class ... {
  date = asDate(() => this.timestamp, (ts) => { this.timestamp = ts })
}
```

which maybe in turn could be simplified into something like

```ts
const asDate = dataTransform((ts) => new Date(ts), (date) => +date)

class ...  extends Model({ timestamp: prop<number>() }) {
  @asDate("timestamp")
  date: Date
}

const date = model.date
model.date = new Date()
```

would something like that be good enough for custom types?

**Comment by xaviergonz** [comment]
Just created a PR with the idea: https://github.com/xaviergonz/mobx-keystone/pull/48

**Comment by terrysahaidak** [comment]
I'm ok with decorator I think. This is a pretty general approach allows creating complex custom types.

**Comment by xaviergonz** [comment]
Published prop transforms as 0.23.0

**Comment by swayf** [comment]
Does it mean @asyncAction from mobx-utils can be used just instead @modelFlow?? =)

**Comment by swayf** [comment]
Time for v1.0.0? =)))

**Comment by alex-shamshurin** [comment]
May be Jan 01 2021 ?
